src/app/appRouter.tsx

```diff
@@ -1,15 +1,34 @@
-import { createBrowserRouter } from 'react-router-dom'
+import { type ReactNode } from 'react'
+import { Navigate, createBrowserRouter } from 'react-router-dom'
 import { featureToggleLoader } from '@/entities/featureToggle'
+import { selectIsAuthorized } from '@/entities/session'
 import { CartPage } from '@/pages/cart'
 import { CategoryPage } from '@/pages/category'
 import { LoginPage } from '@/pages/login'
 import { MainPage } from '@/pages/main'
 import { ProductPage } from '@/pages/product'
 import { WishlistPage } from '@/pages/wishlist'
+import { useAppSelector } from '@/shared/model'
 import { appStore } from './appStore'
 import { baseLayout } from './layouts/baseLayout'
 import { layoutWithSidebar } from './layouts/layoutWithSidebar'
 
+const GuestGuard = ({ children }: { children: ReactNode }) => {
+  const isAuthorized = useAppSelector(selectIsAuthorized)
+
+  if (!isAuthorized) return <Navigate to="/" />
+
+  return <> {children} </>
+}
+
+const AuthGuard = ({ children }: { children: ReactNode }) => {
+  const isAuthorized = useAppSelector(selectIsAuthorized)
+
+  if (isAuthorized) return <Navigate to="/" />
+
+  return <> {children} </>
```

Comment:
Could we not use the fragment?

```suggestion
  return children
```